### PR TITLE
feat: define HostConfig.Devices with a known type in extension-api.d.ts

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Define HostConfig.Devices with a known type in extension-api.d.ts, as defined in 
- https://docs.docker.com/reference/api/engine/version/v1.43/#tag/Container/operation/ContainerCreate
- https://github.com/moby/moby/blob/34950b5ede5fa4dc4e6956655299f26b17b88f51/api/types/container/hostconfig.go#L269-L273

### What issues does this PR fix or reference?

Part of #5989, is needed for #9125 

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
maybe rename the interface to `DeviceMapping`

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -2600,6 +2600,12 @@ declare module '@podman-desktop/api' {
     Options?: { [key: string]: string };
   }
 
+  interface Device {
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
  interface DeviceMapping {
```

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -2653,7 +2659,7 @@ declare module '@podman-desktop/api' {
     CpuQuota?: number;
     CpusetCpus?: string;
     CpusetMems?: string;
-    Devices?: unknown;
+    Devices?: Device[];
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts]:
```suggestion
    Devices?: DeviceMapping[];
```

**Review by benoitf — COMMENTED** [review]
to match the spec

**Review by gastoner — APPROVED** [review]

**Pull request merged**